src/openssl/ssl/server.cr

```diff
@@ -72,6 +72,12 @@ class OpenSSL::SSL::Server
     end
   end
 
+  # This method allows to fetch the SSL SNI string if present
+  def sni : String?
+    host_name = String.new(LibSSL.ssl_get_servername(@ssl, LibSSL::TLSExt::NAMETYPE_host_name))
+    host_name.empty? ? nil : host_name
+  end
```

Comment:
It should check for a NULL pointer instead of instantiating a String then checking it's size:

```crystal
if hostname = LibSSL.ssl_get_servername(@ssl, LibSSL::TLSExt::NAMETYPE_host_name)
  String.new(hostname)
end
```

Reply:
@ysbaddaden done!
keep in mind that `hostname=` will be part of the Client socket (as I answered to @asterite)